src/test/java/com/onaio/steps/handler/actions/TakeSurveyHandlerTest.java

```diff
@@ -161,47 +187,98 @@ public void ShouldNotBeAbleToHandleResultForOtherRequestCode(){
+        TakeSurveyHandlerStub surveyHandler = new TakeSurveyHandlerStub(householdActivitySpy, householdMock, savedForms);
 
-        surveyHandler.handleResult(new Intent().setData(Mockito.mock(Uri.class)), Activity.RESULT_OK);
+        surveyHandler.handleResult(new Intent().setData(mock(Uri.class)), Activity.RESULT_OK);
 
         Mockito.verify(householdMock).setStatus(InterviewStatus.DONE);
-        Mockito.verify(householdMock).update(Mockito.any(DatabaseHelper.class));
+        Mockito.verify(householdMock).update(any(DatabaseHelper.class));
     }
 
     @Test
     public void ShouldUpdateStatusAsInProgressWhenFormHasStatusOfInComplete(){
         mockSavedForm(savedForms, "incomplete");
-        TakeSurveyHandlerStub surveyHandler = new TakeSurveyHandlerStub(householdActivityMock, householdMock, savedForms);
+        TakeSurveyHandlerStub surveyHandler = new TakeSurveyHandlerStub(householdActivitySpy, householdMock, savedForms);
 
-        surveyHandler.handleResult(new Intent().setData(Mockito.mock(Uri.class)), Activity.RESULT_OK);
+        surveyHandler.handleResult(new Intent().setData(mock(Uri.class)), Activity.RESULT_OK);
 
         Mockito.verify(householdMock).setStatus(InterviewStatus.INCOMPLETE);
-        Mockito.verify(householdMock).update(Mockito.any(DatabaseHelper.class));
+        Mockito.verify(householdMock).update(any(DatabaseHelper.class));
     }
 
     private void mockSavedForm(ArrayList<IForm> savedForms, String formStatus) {
-        ODKSavedForm savedForm = Mockito.mock(ODKSavedForm.class);
+        ODKSavedForm savedForm = mock(ODKSavedForm.class);
         savedForms.add(savedForm);
-        Mockito.when(savedForm.getStatus()).thenReturn(formStatus);
+        when(savedForm.getStatus()).thenReturn(formStatus);
     }
 
     @Test
     public void ShouldNotHandleResultForErrorResult(){
         takeSurveyHandler.handleResult(null, Activity.RESULT_CANCELED);
 
         Mockito.verify(householdMock,Mockito.never()).setStatus(InterviewStatus.DONE);
-        Mockito.verify(householdMock,Mockito.never()).update(Mockito.any(DatabaseHelper.class));
+        Mockito.verify(householdMock,Mockito.never()).update(any(DatabaseHelper.class));
+    }
+
+    @Test
+    public void testOpenShouldCallStrategyOpenMethodAndHandleException() throws IOException {
+        doNothing().when(takeSurveyForHouseholdStrategy).open();
+        takeSurveyHandler.open();
+
+        verify(takeSurveyForHouseholdStrategy, times(1)).open();
+
+        // handle exception
+        doThrow(NullPointerException.class).when(takeSurveyForHouseholdStrategy).open();
+        takeSurveyHandler.open();
+
+        ArgumentCaptor<Exception> arException = ArgumentCaptor.forClass(Exception.class);
+        verify(exceptionHandler, times(1)).handle(arException.capture());
+
+        assertEquals(NullPointerException.class.getName(), arException.getValue().getClass().getName());
```

Comment:
Split these into multiple tests so that each fails independently
- Verifying that `takeSurveyForHouseholdStrategy.open()` is called
- Verifying that the exception handler is called with the correct exception

Reply:
done